za_hansard/management/commands/za_hansard_q_and_a_scraper.py

```diff
@@ -239,6 +302,148 @@ def scrape_answers(self, start_url_a, *args, **options):
             if options['limit'] and count >= options['limit']:
                 break
 
+    def handle_api_question_and_reply(self, data):
+        if 'source_file' not in data:
+            print "Skipping {0} due to a missing source_file".format(data['url'])
+            return
+        house = {
+            'National Assembly': 'N'
+        }[data['house']['name']]
+        answer_type = {
+            'written': 'W'
+        }[data['answer_type']]
+        askedby_name = ''
+        askedby_pa_url = ''
+        if 'asked_by_member' in data:
+            askedby_name = data['asked_by_member']['name']
+            if 'pa_url' in data['asked_by_member']:
+                askedby_pa_url = data['asked_by_member']['pa_url']
+        question_text = data['question']
+        # 'code' is one of those like: "NW3847", but in the Code4SA /
+        # PMG API data that's just composed (say) of 'NW' + the
+        # written_number; they don't match with the NW codes that
+        # questions already have in our data. So we can't use the
+        # 'code' field for checking if the question already exists, so
+        # try to identify a unique question by the number + year +
+        # house. (The numbers reset each year.)
+        year = data['date'][:4]
+        # Add whatever number there is to the query:
+        number_q_kwargs = {}
+        number_found = False
+        for filter_key, api_key in (
+                ('written_number', 'written_number'),
+                ('oral_number', 'oral_number'),
+                ('president_number', 'president_number'),
+                ('dp_number', 'deputy_president_number'),
+        ):
+            if data[api_key]:
+                number_found = True
+                number_q_kwargs[filter_key] = data[api_key]
+        existing_kwargs = {'date__year': year, 'house': house}
+        existing_kwargs.update(number_q_kwargs)
+        if not number_found:
+            # We won't be able to accurately tell whether a question
+            # already exists if we don't have one of these number, so
+            # ignore the question and answer completely in that
+            # case. (This is a rare occurence.)
+            print "Skipping {0} because no number was found".format(data['url'])
+            return
+        try:
+            question = find_matching_question(
```

Comment:
As I mentioned in stand-up, this was completely wrong-headed - I'm not sure how I got confused about this. There's a unique constraint that means that there can't be more than one question or answer with the same `written_answer`, `house` and `year`. So, I've removed all of that code.